core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -156,12 +160,19 @@ private[spark] class Executor(
       serializedTask: ByteBuffer)
     extends Runnable {
 
+    /** Whether this task has been killed. */
     @volatile private var killed = false
-    @volatile var task: Task[Any] = _
-    @volatile var attemptedTask: Option[Task[Any]] = None
```

Comment:
This `attemptedTask` vs `task` stuff in the old code is/was really confusing, so thanks for cleaning it up.